ganalytics.cpp

```diff
@@ -473,10 +501,12 @@ GAnalytics::GAnalytics(const QString &trackingID, QObject *parent)
 }
 
 /**
- * Destructor of class GAnalytics.
+ * Destructor of class GAnalytics. Ends the session.
  */
 GAnalytics::~GAnalytics()
 {
+    if (!trackingID().isEmpty())
+        endSession();
```

Comment:
I completely agree that the constructor should not wait for the events to be sent, and that this should happen in an asynchronous manner. My suggestion was only for the destructor to immediately initiate sending events and then wait until all the events are sent, and not saving them somewhere.

Saving this information and loading it the next time the application is used will not help really, since the events which are older than 4 hours are not sent anyway, and usually a user will open a closed application a day (or more) later.

However, as I understand you also want to give users the control to immediately stop and not finish sending any events to google. In this case, maybe a flag would help, like "setSendAllEventsUponQuit(bool enabled)" which can control this behaviour? Or a function "SendAllItemsInQueue()" and a signal "queueEmpty"?

What is the proper way of sending the remaining events with the current API? Setting the send timer interval really low, sending a dummy event, and waiting until the issending signal goes from true to false? That sounds quite cumbersome...